runtime/autoload/health.vim

```diff
@@ -1,433 +1,181 @@
 
-
-" Echo wrapped notes
-function! s:echo_notes(notes) abort
-  if empty(a:notes)
-    return
+""
+" This function represents a check that has not gone correctly within a report
+" However, even with the unsuccesful check, it makes sense to continue
+" checking other health items. Use {msg} to represent the failed health check
+" and optionally a list of suggestions on how to fix it.
+function! health#report_warn(msg, ...) abort
+  if a:0 > 0 && type(a:1) == type([])
+    echo s:format_report_message('WARNING', a:msg, a:1)
+  else
+    echo s:format_report_message('WARNING', a:msg)
   endif
-
-  echo '  Messages:'
-  for msg in a:notes
-    if msg =~# "\n"
-      let msg_lines = []
-      for msgl in filter(split(msg, "\n"), 'v:val !~# ''^\s*$''')
-        call extend(msg_lines, s:textwrap(msgl, 74))
-      endfor
-    else
-      let msg_lines = s:textwrap(msg, 74)
-    endif
-
-    if !len(msg_lines)
-      continue
-    endif
-    echo '    *' msg_lines[0]
-    if len(msg_lines) > 1
-      echo join(map(msg_lines[1:], '"      ".v:val'), "\n")
-    endif
-  endfor
 endfunction
 
-
-" Load the remote plugin manifest file and check for unregistered plugins
-function! s:diagnose_manifest() abort
-  echo 'Checking: Remote Plugins'
-  let existing_rplugins = {}
-
-  for item in remote#host#PluginsForHost('python')
-    let existing_rplugins[item.path] = 'python'
-  endfor
-
-  for item in remote#host#PluginsForHost('python3')
-    let existing_rplugins[item.path] = 'python3'
-  endfor
-
-  let require_update = 0
-  let notes = []
-
-  for path in map(split(&rtp, ','), 'resolve(v:val)')
-    let python_glob = glob(path.'/rplugin/python*', 1, 1)
-    if empty(python_glob)
-      continue
-    endif
-
-    let python_dir = python_glob[0]
-    let python_version = fnamemodify(python_dir, ':t')
-
-    for script in glob(python_dir.'/*.py', 1, 1)
-          \ + glob(python_dir.'/*/__init__.py', 1, 1)
-      let contents = join(readfile(script))
-      if contents =~# '\<\%(from\|import\)\s\+neovim\>'
-        if script =~# '/__init__\.py$'
-          let script = fnamemodify(script, ':h')
-        endif
-
-        if !has_key(existing_rplugins, script)
-          let msg = printf('"%s" is not registered.', fnamemodify(path, ':t'))
-          if python_version == 'pythonx'
-            if !has('python2') && !has('python3')
-              let msg .= ' (python2 and python3 not available)'
-            endif
-          elseif !has(python_version)
-            let msg .= printf(' (%s not available)', python_version)
-          else
-            let require_update = 1
-          endif
-
-          call add(notes, msg)
-        endif
-
-        break
-      endif
-    endfor
-  endfor
-
-  echo '  Status: '
-  if require_update
-    echon 'Out of date'
-    call add(notes, 'Run :UpdateRemotePlugins')
+""
+" This function represents a check that has failed and because of it does not
+" make sense to continue testing the health of the health checker.
```

Comment:
I'm confused about the mention of "does not make sense to continue". Why doesn't it make sense? Why can't a checker report multiple errors?